### PR TITLE
doc: add http api doc

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
add http api doc

### Check List 

Tests 

 - No Code

### CODE REVIEW

**Comment by leoppro** [comment]
PTAL @july2993 @WangXiangUSTC @ericsyh

File: docs/binlog_http_api.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+       },
+       "CommitTS":412518831849473230,
+       "Checkpoint":{
+   
+       },
+       "ErrMsg":""
+   }
+    ```
+
+1. Get all metrics of Pump
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
2. Get all metrics of Pump
```

File: docs/binlog_http_api.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    ```shell
+    curl http://{PumpIP}:8250/metrics
+    ```
+
+1. Get the status of all drainers
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
3. Get the status of all drainers
```

File: docs/binlog_http_api.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+           "state":"paused",
+           "isAlive":false,
+           "score":0,
+           "label":null,
+           "maxCommitTS":412361808537191540,
+           "updateTS":412518474704487076
+       }
+   ]
+    ```
+1. Change the Pump status
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
4. Change the Pump status
```

File: docs/binlog_http_api.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    ```shell
+    $curl -X PUT http://127.0.0.1:8250/state/ip-172-16-5-71:8250/stop
+    
+    {
+        "message":"apply action stop success!",
+        "code":200
+    }
+    ```
+   
+1. Get the binlog by TS
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
5. Get the binlog by TS
```

File: docs/binlog_http_api.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+   
+   ```shell
+   $curl http://127.0.0.1:8250/debug/binlog/412518831548007863
+   
+    tp:Commit start_ts:412518831548007786 commit_ts:412518831548007863 prewrite_key:"t\200\000\000\000\000\000-\212_i\200\000\000\000\000\000\000\001\003\200\000\000\000\000\teK\003\200\000\000\000\000\007\230L"
+    
+     GetMvccByEncodedKey:
+   ```
+
+1. Start the GC
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
6. Start the GC
```

File: docs/binlog_http_api.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+            "ip-172-16-5-74:8250":412361808550297951,
+            "ip-172-16-5-75:8250":412361808550297952
+        },
+        "Synced":true,
+        "LastTS":412361808537191540,
+        "TsMap":""
+    }
+   ```
+   
+1. Get all metrics of Drainer
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
2. Get all metrics of Drainer
```

File: docs/binlog_http_api.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+    ```shell
+    curl http://{DrainerIP}:8249/metrics
+    ```
+
+1. Get the lastest commit ts of Drainer
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
3. Get the lastest commit ts of Drainer
```

File: docs/binlog_http_api.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    {
+      "message": "get drainer's latest ts success!",
+      "code": 200,
+      "data": {
+        "ts": 412361808537191540
+      }
+    }
+   ```
+
+1. Change the Drainer status
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
4. Change the Drainer status
```

File: docs/binlog_http_api.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
+# TiDB-Binlog HTTP API
+
+## Pump
+
+`PumpIP` is the ip of the Pump server. `8250` is the default port of Pump.
+
+1. Get the current status of Pump
```
Comment (ericsyh) [inline docs/binlog_http_api.md]:
```suggestion
1. Get the current status of Pump servers
```

File: docs/binlog_http_api.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+           "updateTS":412518474704487076
+       }
+   ]
+    ```
+1. Change the Pump status
+
+    `NodeID` is the node id of the Pump server. `Action` is the action to execute[possible values: `pause`, `close`].
+
+    ```shell
+    curl -X PUT http://{PumpIP}:8250/state/{NodeID}/{Action}
```
Comment (ericsyh) [inline docs/binlog_http_api.md]: Question here:  in binlogctl we have `pause-pump`, `offline-pump` and `update-pump` actions while seems HTTP API only have `pause` and `close` ?

File: docs/binlog_http_api.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+      "code": 200,
+      "data": {
+        "ts": 412361808537191540
+      }
+    }
+   ```
+
+1. Change the Drainer status
+
+    `NodeID` is the node id of the Pump server. `Action` is the action to execute[possible values: `pause`, `close`].
```
Comment (ericsyh) [inline docs/binlog_http_api.md]: Same action question here

**Comment by GregoryIan** [comment]
@leoppro please address comments

File: docs/binlog_http_api.md (hunk 10 above)
Comment (GregoryIan) [reply docs/binlog_http_api.md]: yep, what's meaning of `close`? `offline`?

File: docs/binlog_http_api.md (hunk 1 above)
Comment (leoppro) [reply docs/binlog_http_api.md]: `1.` is the list mark in markdown. The markdown renderer will evaluate the correct number for the list items.
you can see: https://github.com/pingcap/tidb-binlog/pull/805/files?short_path=9d85a25#diff-9d85a258cba937911ccf00980402cef8

File: docs/binlog_http_api.md (hunk 10 above)
Comment (leoppro) [reply docs/binlog_http_api.md]: `pause` is equivalent to `pause-pump\pause-drainer`, close is equivalent to `offline-pump\offline-drainer`.
the function of `update` are not supported by HTTP API.

File: docs/binlog_http_api.md (hunk 1 above)
Comment (ericsyh) [reply docs/binlog_http_api.md]: ok, it's fine.

**Pull request merged**